### ISSUE TITLE
simplify installation process

### ISSUE THREAD

**Issue by GliderGeek** [GliderGeek/pocket2rm#7]
currently difficult for someone without programming knowledge. potential improvements:
- precompile for target
- scp from go. https://github.com/bramvdbogaerde/go-scp might be interesting
- popup for key instead of commandline entry

**Comment by lakano** [comment]
And this could prevent to take too much space for golang/libs on the very small disk of rM ! :)

**Comment by faroit** [comment]
I wonder if it would be easier to just send via the remarkable cloud and use rmapi to send the docs?

**Comment by GliderGeek** [comment]
i think it depends on your needs. the reason i like pocket a lot is that there are existing integrations for my other devices (and i already had an account there):

- iphone shortcut for browser
- browser extension for firefox on laptop
- official app on kobo ereader
- this tool on remarkable (this didn't exist so i created it 😄 )

getting this to work on my phone with a push to remarkable cloud is a bit difficult i think.

**Comment by faroit** [comment]
Not clear if I understand what you are saying: 
Are you running this tool on your phone? I guess not.
What I meant its replacing the ssh connection within this package with rmapi or rmapy to simplify installation process on the backend side. This could also be additional. But using the rm cloud to send files seems more flexible to me.

**Comment by GliderGeek** [comment]
ok, i misinterpreted your comment. interesting idea, will think about it.

**Comment by r4v5** [comment]
Would you be willing to use github actions / github releases to build a pre-compiled version of the ARM binary? Downside is you'd also probably need to start versioning them, which can be a hassle.

**Comment by GliderGeek** [comment]
Yeah i would be open to that. Would that help a lot? I guess it would be mainly helpful if all go binaries were present (also pocket2rm-setup for windows/mac/linux such that it’s no longer needed to compile manually?

**Comment by GliderGeek** [comment]
@faroit: getting rid of `scp` would be beneficial as i think that this is not available by default on windows?

regarding rmapi: do you know if it's possible to place files in a random place on the device using this? from glancing over the repo, it seems that this is only works for placing files inside the "My files" part of the device (`/home/root/.local/share/remarkable/xochitl/`).

i need to place files in:
- `/home/root/`
- `/etc/systemd/system/`

do you know if this works with `rmapi`?

**Comment by stiliajohny** [comment]
hi @GliderGeek and all, 
I have prepared a bash install script.. ( push and PR comes up soon ) 

Would you consider having a script to do all the install?

also on your point to have a popup, it will have a dependency on a cli app like `zenity` I would recommend ( as I have done ) within the script to ask for it  and pass it to the go app like a variable 
eg: main --key RGTHFVBKJLUOIT

I know in python its called argparser not sure in go :D

**Comment by GliderGeek** [comment]
in the meantime the installation has been significantly simplified by the introduction of a single install script. it would still be nice to make this an order of magnitude easier, such that non-devs can also use this.

**Comment by RhvdG** [comment]
I'd like to install this, but the instructions are not clear to me. Do I need to install golang? What does 'clone or download this repository' mean? I'd also like to see an uninstall 'magic phrase', just in case. To give you an idea of my competence using the command line: I can ssh into the device, I installed the ddvk hacks without any trouble.

**Comment by GliderGeek** [comment]
> Do I need to install golang

yes

> What does 'clone or download this repository' mean?

the green button on the main repository page. if you dont know what cloning means, i would go for download.

<img width="939" alt="Screenshot 2021-02-23 at 08 53 28" src="https://user-images.githubusercontent.com/16109193/108815367-a3f15f80-75b4-11eb-99fb-c52e0cfc41a5.png">

> I'd also like to see an uninstall 'magic phrase', just in case

There is none at the moment. There are a few files copied to your device we you should manually remove to uninstall.

I understand that this might be a bit intimidating when you're not familiar with software development.
